internal/datastore/crdb/crdb.go

```diff
@@ -577,24 +577,26 @@ func (cds *crdbDatastore) features(ctx context.Context) (*datastore.Features, er
 		return nil, err
 	}
 
-	// streams don't return at all if they succeed, so the only way to know
-	// it was created successfully is to wait a bit and then cancel
-	streamCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
-	time.AfterFunc(1*time.Second, cancel)
-
-	_ = cds.writePool.ExecFunc(streamCtx, func(ctx context.Context, tag pgconn.CommandTag, err error) error {
-		if err != nil && errors.Is(err, context.Canceled) {
-			features.Watch.Status = datastore.FeatureSupported
-			features.Watch.Reason = ""
-		} else if err != nil {
-			features.Watch.Status = datastore.FeatureUnsupported
-			features.Watch.Reason = fmt.Sprintf("Range feeds must be enabled in CockroachDB and the user must have permission to create them in order to enable the Watch API: %s", err.Error())
+	// Start a changefeed with an invalid value. If we get back an invalid value error (SQLSTATE 22023)
+	// then we know that the datastore supports watch. If we get back any other error, then we know that
+	// the datastore does not support watch emits or there is a permissions issue.
+	_ = cds.writePool.ExecFunc(ctx, func(ctx context.Context, tag pgconn.CommandTag, err error) error {
```

Comment:
The changestream does not run through any connection pools based on a recommendation from CRL; why is it being done that way here?

not sure if this runs in a loop or a certain cadence, but that would cause connections to be terminated (because of the error)

Reply:
Because its going to fail either way; the reason to not use a pool is because it can be a long-running operation

Reply:
Its only called on startup, so it should be fine